packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -529,35 +553,169 @@ class RenderListWheelViewport
     size = constraints.biggest;
   }
 
+  /// Get the index of a child by looking at its parentData.
+  int indexOf(RenderBox child) {
+    assert(child != null);
+    final ListWheelParentData childParentData = child.parentData;
+    assert(childParentData.index != null);
+    return childParentData.index;
+  }
+
+  /// Return the index of the child that should be at the given offset.
+  int scrollOffsetToIndex(double scrollOffset) => (scrollOffset / itemExtent).floor();
+
+  /// Return the scroll offset of the child with the given index.
+  double indexToScrollOffset(int index) => index * itemExtent;
+
+  // Create a child at the given index.
+  void _createChild(int index, {RenderBox after}) {
+    invokeLayoutCallback<BoxConstraints>((BoxConstraints constraints) {
+      assert(constraints == this.constraints);
+      childManager.createChild(index, after: after);
+    });
+  }
+
+  void _destroyChild(RenderBox child) {
+    invokeLayoutCallback<BoxConstraints>((BoxConstraints constraints) {
+      assert(constraints == this.constraints);
+      childManager.removeChild(child);
+    });
+  }
+
+  void _layoutChild(RenderBox child, BoxConstraints constraints, int index) {
+    child.layout(constraints, parentUsesSize: true);
+    final ListWheelParentData childParentData = child.parentData;
+    final double crossPosition = size.width / 2.0 - child.size.width / 2.0;
+    childParentData.offset = new Offset(crossPosition, indexToScrollOffset(index));
+  }
+
+  /// Layout base on children provided by [childManager].
+  ///
+  /// From the current scroll offset, we can calculate the minimum index and
+  /// maximum index that is visible in the viewport. We can also get the index
+  /// range of the children that are currently active.
+  /// We will modify the child list so that children that are not visible
+  /// will be removed, and those that should visible but not yet created will be
+  /// instantiate by the childManager.
   @override
   void performLayout() {
-    double currentOffset = 0.0;
+    final BoxConstraints childConstraints =
+      constraints.copyWith(
+        minHeight: _itemExtent,
+        maxHeight: _itemExtent,
+        minWidth: 0.0,
+      );
+
+    // The height, in pixel, that children will be visible and might be laid out
+    // and painted.
+    double visibleHeight = size.height;
+    // If [renderChildrenOutsideViewport] is true, we spawn a redundant amount
+    // of children, those that are in the backside of the cylinder won't be
+    // painted anyway.
+    if (renderChildrenOutsideViewport)
+      visibleHeight *= 2;
+
+    final double firstVisibleOffset =
+        offset.pixels + _itemExtent / 2 - visibleHeight / 2;
+    final double lastVisibleOffset = firstVisibleOffset + visibleHeight;
+
+    // The index range that we want to spawn children. We find indexes that
+    // are in the interval [firstVisibleOffset, lastVisibleOffset).
+    int targetFirstIndex = scrollOffsetToIndex(firstVisibleOffset);
```

Comment:
I think it would be useful to explain what the difference between the target child list and the current child list.

What do the lists represent?  What does one contain that the other doesn't?

That would be useful for understanding the logic farther down where you talk about different cases in these two lists.

Reply:
I've modified the description of performLayout() to be more clear about this.